src/constructs/lambda/lambda.ts

```diff
@@ -21,12 +22,28 @@ interface GuFunctionProps extends Omit<FunctionProps, "code"> {
   apis?: ApiProps[];
 }
 
+function defaultMemorySize(runtime: Runtime, memorySize?: number): number {
+  if (memorySize) {
+    return memorySize;
+  } else {
+    switch (runtime.family) {
+      case RuntimeFamily.JAVA:
+        return 1024;
+      default:
+        return 512;
```

Comment:
I think the other main family that we want to cater for is `NODE`, so I've based this on https://github.com/guardian/tag-janitor/blob/main/cdk/lib/cdk-stack.ts#L54 and https://github.com/guardian/google-chat-bots/blob/main/cdk/lib/constants.ts#L29.